include/ews/ews.hpp

```diff
@@ -9429,238 +9429,1440 @@ namespace ews
+    {
+        inline std::string enum_to_str(email_address_key k)
+        {
+            switch (k)
+            {
+            case email_address_key::email_address_1:
+                return "EmailAddress1";
+            case email_address_key::email_address_2:
+                return "EmailAddress2";
+            case email_address_key::email_address_3:
+                return "EmailAddress3";
+            default:
+                throw exception("Bad enum value");
+            }
+        }
+    }
+
+    class email_address final
+    {
+    public:
+        explicit email_address(email_address_key k, std::string value)
+            : key_(std::move(k)), value_(std::move(value))
+        {
+        }
+
+        static email_address
+        from_xml_element(const rapidxml::xml_node<char>& node)
+        {
+            using rapidxml::internal::compare;
+
+            // <t:EmailAddresses>
+            //  <Entry Key="EmailAddress1">donald.duck@duckburg.de</Entry>
+            //  <Entry Key="EmailAddress3">dragonmaster1999@yahoo.com</Entry>
+            // </t:EmailAddresses>
+
+            EWS_ASSERT(compare(node.local_name(), node.local_name_size(),
+                               "Entry", std::strlen("Entry")));
+            auto key = node.first_attribute("Key");
+            EWS_ASSERT(key && "Expected attribute Key");
+            return email_address(
+                str_to_key(std::string(key->value(), key->value_size())),
+                std::string(node.value(), node.value_size()));
+        }
+
+        std::string to_xml() const
+        {
+            std::stringstream sstr;
+            sstr << " <t:"
+                 << "EmailAddresses"
+                 << ">";
+            sstr << " <t:Entry Key=";
+            sstr << "\"" << internal::enum_to_str(get_key());
+            sstr << "\">";
+            sstr << get_value();
+            sstr << "</t:Entry>";
+            sstr << " </t:"
+                 << "EmailAddresses"
+                 << ">";
+            return sstr.str();
+        }
+
+        email_address_key get_key() const { return key_; }
+        const std::string& get_value() const EWS_NOEXCEPT { return value_; }
+
+    private:
+        email_address_key key_;
+        std::string value_;
+        friend bool operator==(const email_address&, const email_address&);
+        static email_address_key str_to_key(const std::string& keystring)
+        {
+            email_address_key k;
+            if (keystring == "EmailAddress1")
+            {
+                k = email_address_key::email_address_1;
+            }
+            else if (keystring == "EmailAddress2")
+            {
+                k = email_address_key::email_address_2;
+            }
+            else if (keystring == "EmailAddress3")
+            {
+                k = email_address_key::email_address_3;
+            }
+            else
+            {
+                throw exception(std::string("Unrecognized key: ") + keystring);
```

Comment:
Why do you wrap this with an explicit call to std::string constructor? Remove this. There is a `operator(const char*, const std::string&)` in the standard for this.

Here and other locations.